atom/browser/atom_permission_manager.cc

```diff
@@ -155,9 +156,15 @@ int AtomPermissionManager::RequestPermissionsWithDetails(
     return content::PermissionController::kNoPendingOperation;
   }
 
+  bool secure_mode_enabled = Browser::Get()->secure_mode_enabled();
+
   if (request_handler_.is_null()) {
     std::vector<blink::mojom::PermissionStatus> statuses;
     for (auto permission : permissions) {
+      if (secure_mode_enabled) {
+        statuses.push_back(blink::mojom::PermissionStatus::DENIED);
```

Comment:
There are also permission checks (not requests) that we should `return false` for when secure mode is enabled to be consistent here